### PR TITLE
basic validation fixes

### PR DESCRIPTION
**Pull request by Zenen-freshworks**

### CODE REVIEW

**Review by mauriciofreshw — DISMISSED** [review]

File: server/validation.js [hunk 1]
```diff
@@ -238,8 +238,8 @@ const ParticipantBatchSchema = yup.array().of(
       // Contact info
       firstName: yup.string().required(errorMessageIndex(index, indexName)),
       lastName: yup.string().required(errorMessageIndex(index, indexName)),
-      phoneNumber: yup.mixed().required(errorMessageIndex(index, indexName)),
-      emailAddress: yup.string().required(errorMessageIndex(index, indexName)).matches(/^(.+@.+\..+)?$/, `Invalid email address (row ${index})`),
+      phoneNumber: yup.string().matches(/^([0-9]{10})$/, { excludeEmptyString: true, message: `Phone number must be provided as 10 digits (index ${index})` }).nullable(errorMessageIndex(index)),
```
Comment (fwkendall) [inline server/validation.js:241]: This should be required, not nullable I think?

Comment (fwkendall) [reply server/validation.js:241]: I think we want to follow what's used for single participant POST vie the PEOI form:

`phoneNumber: yup.string().required(errorMessage).matches(/^\d{10}$/, 'Phone number must be provided as 10 digits'),`

File: server/validation.js [hunk 2]
```diff
@@ -348,7 +348,7 @@ const ExternalHiredParticipantSchema = yup.object().shape({
   firstName: yup.string().required('First Name is required'),
   lastName: yup.string().required('Last Name is required'),
   phoneNumber: yup.string().nullable(errorMessage).matches(/^[0-9]{10}$/, 'Phone number must be provided as 10 digits'),
```
Comment (fwkendall) [inline server/validation.js:350]: should be required

File: server/validation.js [hunk 3]
```diff
@@ -377,7 +377,7 @@ const ParticipantEditSchema = yup.object().noUnknown('Unknown field in entry').s
...
```
Comment (fwkendall) [inline server/validation.js:379]: should be required

**Comment by fwkendall** [comment]
I think the client validation has some inconsistencies too 😄

File: server/validation.js [hunk 4]
```diff
@@ -238,8 +238,8 @@ const ParticipantBatchSchema = yup.array().of(
       // Contact info
       firstName: yup.string().required(errorMessageIndex(index, indexName)),
       lastName: yup.string().required(errorMessageIndex(index, indexName)),
-      phoneNumber: yup.mixed().required(errorMessageIndex(index, indexName)),
-      emailAddress: yup.string().required(errorMessageIndex(index, indexName)).matches(/^(.+@.+\..+)?$/, `Invalid email address (row ${index})`),
+      phoneNumber: yup.string().matches(/^([0-9]{10})$/, { excludeEmptyString: true, message: `Phone number must be provided as 10 digits (index ${index})` }).nullable(errorMessageIndex(index)),
+      emailAddress: yup.string().email(`should be a valid email address (index ${index})`).nullable(errorMessageIndex(index)),
```
Comment (fwkendall) [inline server/validation.js:242]: should be required

**Review by mauriciofreshw — DISMISSED** [review]

File: server/validation.js (hunk 4 above)
Comment (ashtonmeuser) [reply server/validation.js:242]: Eagle eyes

File: server/validation.js [hunk 5]
```diff
@@ -238,8 +238,8 @@ const ParticipantBatchSchema = yup.array().of(
       // Contact info
       firstName: yup.string().required(errorMessageIndex(index, indexName)),
       lastName: yup.string().required(errorMessageIndex(index, indexName)),
-      phoneNumber: yup.mixed().required(errorMessageIndex(index, indexName)),
-      emailAddress: yup.string().required(errorMessageIndex(index, indexName)).matches(/^(.+@.+\..+)?$/, `Invalid email address (row ${index})`),
+      phoneNumber: yup.string().required(errorMessage).matches(/^\d{10}$/, 'Phone number must be provided as 10 digits'),
+      emailAddress: yup.string().email(`should be a valid email address (index ${index})`).nullable(errorMessageIndex(index)),
```
Comment (ashtonmeuser) [inline server/validation.js:242]:
```suggestion
      emailAddress: yup.string().required(errorMessage).email(`should be a valid email address (index ${index})`).nullable(errorMessageIndex(index)),
```

File: server/validation.js (hunk 4 above)
Comment (fwkendall) [reply server/validation.js:242]: I think this is still applicable for the email, everything else looks great :)

**Review by fwkendall — APPROVED** [review]

**Review by mauriciofreshw — APPROVED** [review]

**Pull request merged**